zvariant_derive/tests/tests.rs

```diff
@@ -59,6 +59,22 @@ fn derive_enum_no_repr() {
     assert_eq!(TestEnum::try_from(Value::U32(3)), Err(Error::IncorrectType));
     assert_eq!(TestEnum::try_from(Value::Str("Foo".into())), Err(Error::IncorrectType));
 }
+
+#[test]
+fn derive_enum_string() {
```

Comment:
Kudos for adding tests. :+1: However, I'd prefer you modify the existing tests in `zvariant/src/lib.rs`. The `derive` test case there already covers enums as strings. You can just add the Value bits to the same enum(s).